src/core/pointcloud/qgspointcloudstatscalculator.cpp

```diff
@@ -0,0 +1,268 @@
+/***************************************************************************
+                         qgspointcloudstatscalculator.cpp
+                         --------------------
+    begin                : April 2022
+    copyright            : (C) 2022 by Belgacem Nedjima
+    email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgspointcloudstatscalculator.h"
+
+#include "qgspointcloudindex.h"
+#include "qgsmessagelog.h"
+#include "qgspointcloudattribute.h"
+#include "qgspointcloudrequest.h"
+
+#include "qgspointcloudrenderer.h"
+
+#include "qgsapplication.h"
+#include "qgspointcloudstatscalculationtask.h"
+#include "qgsfeedback.h"
+#include "qgspointcloudblockrequest.h"
+
+#include <QQueue>
+#include <QtConcurrent/QtConcurrentMap>
+
+struct StatsProcessor
+{
+    typedef QMap<QString, QgsPointCloudStatsCalculator::AttributeStatistics> result_type;
+
+    StatsProcessor( QgsPointCloudIndex *index, QgsPointCloudRequest request, QgsFeedback *feedback )
+      : mIndex( index ), mRequest( request ), mFeedback( feedback )
+    {
+
+    }
+
+    QMap<QString, QgsPointCloudStatsCalculator::AttributeStatistics> operator()( IndexedPointCloudNode node )
+    {
+      QMap<QString, QgsPointCloudStatsCalculator::AttributeStatistics> statsMap;
```

Comment:
it would be more efficient not to use QString (and QMap) for stats - how about simply QVector<AttributeStatistics> where each item would be for one requested attribute...

Reply:
I don't see a problem with returning the statistics as a map since the number of attributes will be way too small for this optimization to make sense, and it will be really inconvenient to check what the actual returned attributes are (especially in case the requested attributes don't match with the available attributes in the dataset)...